### ISSUE TITLE
Support for named parameters in mappings (Solidity v0.8.18)

### ISSUE THREAD

**Issue by PaulRBerg** [protofire/solhint#397]
Solidity v0.8.18 was [released](https://blog.soliditylang.org/2023/02/01/solidity-0.8.18-release-announcement/) earlier this week, and one of the newly introduced features are [named parameters in mappings](https://twitter.com/PaulRBerg/status/1621949932608360453). That means that we can now define mappings like this:

```solidity
contract Foo {
    mapping(address key => uint256 value) public items;
    mapping(string name => uint256 balance) public users;
}
```

Solhint throws with the following error when it sees the code above:

> Linter: Parse error: extraneous input 'key' expecting '=>' [undefined]

**Comment by fvictorio** [comment]
Two things to do here:

1) Update the parser (I already added support for named parameters in it)
2) Open an issue about adding a new rule that suggests using named parameters. This rule will only make sense for solidity `>=0.8.18`, so it should be disabled by default, and not included in the recommended preset config.

**Comment by PaulRBerg** [comment]
Agree, a rule would be cool. But I think until the issue gets implemented, just adding support for the parsing would be helpful.

**Comment by andreivladbrg** [comment]
Hey @fvictorio, do you have any updates on this? My github actions are failing because of this.

**Comment by cyberhorsey** [comment]
Also would love this : )

**Comment by dbale-altoros** [comment]
working on this update

**Comment by dbale-altoros** [comment]
Guys, I created a new rule for this in this PR
https://github.com/protofire/solhint/pull/403

**Comment by PaulRBerg** [comment]
Looks like this issue can be closed now, since v3.4.0 includes support for named mapping params 🥳

https://github.com/protofire/solhint/releases/tag/v3.4.0
